### PR TITLE
Episode popScene MarkItemWatched

### LINKED ISSUE #616: "Refresh" episode list when reappearing on screen
**Describe the bug**
I noticed and reproduced a few times that after starting an unwatched episode from the episode list of a season, and letting playback complete, the checkmark does not appear.

**To Reproduce**

1. Go to the episode list page of a season of a show
2. Find an unwatched episode
3. Play it, allow it to finish
4. Now that you're back at the season page, see that the checkmark has not appeared
5. Press back to return to the show page
6. Click to enter that season again
7. Scroll to the episode you just watched
8. See that it now has the "watched" checkmark

**Expected behavior**
"Watched" checkmark appears immediately after watching

**Logs**

**Screenshots**

**Additional context**

Current stable release

### PR DESCRIPTION
**Pull request by candry7731**
Added the MarkItemWatched() to the autoPlayNextEpisode function
This marks the episode in the episode list marked as played

### CODE REVIEW

**Comment by candry7731** [comment]
I would like to reload the "Continue Watching" row only but I can not wrap my head around the logic so instead I am reloading the entire home scene.... This removes the "Watched Item" from the home scene.

File: source/VideoPlayer.brs [hunk 1]
```diff
@@ -344,18 +348,22 @@ sub autoPlayNextEpisode(videoID as string, showID as string)
         if data <> invalid and data.Items.Count() = 2
             ' remove finished video node
             m.global.sceneManager.callFunc("popScene")
+            MarkItemWatched(showID)
```
Comment (jimdogx) [inline source/VideoPlayer.brs]: Looking at the logic, I think you only need to call MarkItemWatched() once, right before this "IF" block.  Regardless of whether the user can auto play the next episode or not, the "just finished" video can be marked watched.

**Review by jimdogx — CHANGES_REQUESTED** [review]

**Comment by jimdogx** [comment]
Just tested this, but from the TV Shows page (not the home screen).  When I watch an episode until the end, it auto plays the next episode, but it marks ALL episodes in EVERY season as watched LOL.

**Comment by candry7731** [comment]
Weird, maybe we need to change it back to how it was?

**Comment by jimdogx** [comment]
> Weird, maybe we need to change it back to how it was?

I haven't stepped through the code, but I'm guessing that "showID" is the entire "Show".  What you probably need is just the episode.  It looks like autoplay works by querying for the show and getting two episodes and then playing the second one in the array.  Again, I'm not familiar with this code, but I'm wondering if the first item in the array is the episode just played?

I'm talking about these lines:

```
        if data <> invalid and data.Items.Count() = 2
            ' remove finished video node
            m.global.sceneManager.callFunc("popScene")
            ' setup new video node
            nextVideo = CreateVideoPlayerGroup(data.Items[1].Id)
```

Maybe Items[0].Id is what needs to be marked as watched?  You'd have to step through and investigate.

File: source/VideoPlayer.brs [hunk 2]
```diff
@@ -331,6 +335,7 @@ function getAudioInfo(meta as object) as object
 end function
 
 sub autoPlayNextEpisode(videoID as string, showID as string)
+    MarkItemWatched(showID)
```
Comment (cewert) [inline source/VideoPlayer.brs]:
```suggestion
    MarkItemWatched(videoID)
```
The video that was just watched is a fucntion parameter `sub autoPlayNextEpisode(videoID as string, showID as string)`

Comment (cewert) [reply source/VideoPlayer.brs]: @jimdogx is right about `Items[0].Id` - that is the video ID. But if the api query fails then `Items[0].Id` will be invalid. Use `videoID` instead

File: source/VideoPlayer.brs [hunk 3]
```diff
@@ -55,7 +55,11 @@ sub AddVideoContent(video, mediaSourceId, audio_stream_idx = 1, subtitle_idx = -
                 video.content.watched = not video.content.watched
                 group = m.scene.focusedChild
                 group.timeLastRefresh = CreateObject("roDateTime").AsSeconds()
-                group.callFunc("refresh")
+                m.global.sceneManager.callFunc("clearScenes")
+                group = CreateHomeGroup()
+                group.userConfig = m.user.configuration
+                group.callFunc("loadLibraries")
+                m.global.sceneManager.callFunc("pushScene", group)
```
Comment (cewert) [inline source/VideoPlayer.brs:64]: The home rows not refreshing is a known bug #624 Let's keep this PR focused on "marking episodes as watched" and tackle the refresh bug in another PR

**Comment by cewert** [comment]
> Added the MarkItemWatched() to the autoPlayNextEpisode function
This marks the episode in the episode list marked as played

> Should fix https://github.com/jellyfin/jellyfin-roku/issues/616

To be clear, this PR will not fix #616 because that feature hasn't been implemented yet (refreshing the episode list screen)

**Comment by neilsb** [comment]
Can I just check what problem this is solving?   In the current store client, when I autoplay a series (server 10.8.4) as each episode is completed it is marked as played on the server.   The client reports the playback progress to the server while playing, and when the episode get within a certain %ag of the end, the item is marked as played.

So for me it is working as expected without this PR.  Can anyone reproduce the error of a series being autoplayed and the items not being marked as played on the server using the store version?

**Comment by alanazar** [comment]
> Can I just check what problem this is solving? In the current store client, when I autoplay a series (server 10.8.4) as each episode is completed it is marked as played on the server. The client reports the playback progress to the server while playing, and when the episode get within a certain %ag of the end, the item is marked as played.
> 
> So for me it is working as expected without this PR. Can anyone reproduce the error of a series being autoplayed and the items not being marked as played on the server using the store version?

I wasn't able to replicate this problem either:
Jellyfin server 10.8.4 on Windows
Played the 1st episode of a tv show till the end, then it loaded the 2nd episode
Played the 4th episode till the end, then it loaded the 5th episode

Both 1st and 4th episode are marked as watched
and 2+3 are unwatched (mentioning this because Jimi said it marked all as watched)
the marked watch is only showing on the web client of the server.... the Roku app doesn't seem to show which episodes were watched

**Comment by candry7731** [comment]
Seems like this is working, without this.

**Pull request closed without merging**